src/main/java/org/ow2/petals/deploymentmodel/ModelDeployer.java

```diff
@@ -0,0 +1,175 @@
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+
+import org.apache.commons.io.FileUtils;
+import org.ow2.petals.admin.api.PetalsAdministration;
+import org.ow2.petals.admin.api.PetalsAdministrationFactory;
+import org.ow2.petals.admin.api.artifact.Component.ComponentType;
+import org.ow2.petals.admin.api.artifact.lifecycle.ArtifactLifecycleFactory;
+import org.ow2.petals.admin.api.artifact.lifecycle.ComponentLifecycle;
+import org.ow2.petals.admin.api.artifact.lifecycle.ServiceAssemblyLifecycle;
+import org.ow2.petals.admin.api.exception.ArtifactAdministrationException;
+import org.ow2.petals.admin.api.exception.ArtifactDeployedException;
+import org.ow2.petals.admin.api.exception.ConnectionFailedException;
+import org.ow2.petals.admin.api.exception.ContainerAdministrationException;
+import org.ow2.petals.admin.api.exception.DuplicatedServiceException;
+import org.ow2.petals.admin.api.exception.MissingServiceException;
+import org.ow2.petals.deploymentmodel.busmodel.Bus;
+import org.ow2.petals.deploymentmodel.busmodel.ComponentInstance;
+import org.ow2.petals.deploymentmodel.busmodel.ContainerInstance;
+import org.ow2.petals.deploymentmodel.busmodel.ProvisionedMachine;
+import org.ow2.petals.deploymentmodel.busmodel.ServiceUnitInstance;
+import org.ow2.petals.deploymentmodel.serviceunitmodel.ServiceUnit;
+import org.ow2.petals.jbi.descriptor.JBIDescriptorException;
+import org.ow2.petals.jbi.descriptor.original.JBIDescriptorBuilder;
+import org.ow2.petals.jbi.descriptor.original.generated.Jbi;
+
+public class ModelDeployer {
+
+    private static final Logger LOG = Logger.getLogger(ModelDeployer.class.getName());
+
+    private PetalsAdministration petalsAdmin;
+
+    private ArtifactLifecycleFactory artifactLifecycleFactory;
+
+    private JBIDescriptorBuilder jdb;
+
+    Map<String, ComponentInstance> componentsByName = new HashMap<String, ComponentInstance>();
+
+    private HashSet<String> deployedComponents = new HashSet<String>();
+
+    private HashSet<String> deployedServiceUnits = new HashSet<String>();
+
+    public ModelDeployer() throws DuplicatedServiceException, MissingServiceException, JBIDescriptorException {
+        this(PetalsAdministrationFactory.getInstance().newPetalsAdministrationAPI(), null,
+                JBIDescriptorBuilder.getInstance());
+
+    }
+
+    public ModelDeployer(PetalsAdministration petalsAdmin, ArtifactLifecycleFactory artifactLifecycleFactory,
+            JBIDescriptorBuilder jdb) {
+        this.petalsAdmin = petalsAdmin;
+        if (artifactLifecycleFactory == null) {
+            this.artifactLifecycleFactory = petalsAdmin.newArtifactLifecycleFactory();
+        } else {
+            this.artifactLifecycleFactory = artifactLifecycleFactory;
+        }
+        this.jdb = jdb;
+    }
+
+    public void deployModel(Model model) throws ConnectionFailedException, ContainerAdministrationException,
```

Comment:
I saw a usage as `busModel1.deploy()` instead of using a class `ModelDeployer`. But why not ?